lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -121,6 +121,21 @@ static inline ComplementMatcher<SubType> m_Complement(const SubType &subExpr) {
   return ComplementMatcher<SubType>(subExpr);
 }
 
+/// Return true if the op will be flattened afterwards. Op will be flattend if
+/// it has a single user which has a same op type.
+static bool shouldBeFlattened(Operation *op) {
+  assert((isa<AndOp, OrOp, XorOp, AddOp, MulOp>(op) &&
+          "must be commutative operations"));
+  if (op->hasOneUse()) {
+    auto *user = *op->getUsers().begin();
+    if (user->getName() == op->getName() &&
```

Comment:
extreme nit, not sure if it's even worth changing, but `if (a) return true` can often be `return a`, and I think you could return the condition of the if in this case.